RealmSwift/Sync.swift

```diff
@@ -268,6 +272,25 @@ public struct Functions {
             }
         }
     }
+
+    #if canImport(Combine)
+    /// The implementation of @dynamicMemberLookup that allows for dynamic remote function calls.
+    @available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, macCatalyst 13.0, macCatalystApplicationExtension 13.0, *)
+    public subscript(dynamicMember string: String) -> ([AnyBSON]) -> Future<AnyBSON, Error> {
```

Comment:
would it be possible to reuse the existing subscript method and just wrap the responses with promises? Just to avoid the duplication of logic.

Reply:
Will appreciate any guides on how to make it.

Reply:
```
syncUser?.functions[dynamicMember: "updateUserData"]([["favourite_colour": "green", "apples": 10]]) { _, error  in
            XCTAssertNil(error)
            userDataEx.fulfill()
        }
```

Reply:
This is still doable– saves you the conversion step.